### PR TITLE
add copy function to disassembly views

### PR DESCRIPTION
**Pull request by lievenhey**
This allows the user to copy text from the sourcecodeview and the disassembly view. This is done by reimplementing QTreeView with a custom keyPressEvent function that copies everything from selectionModel() to the clipboard and not just currentIndex like the default implementation does.

### CODE REVIEW

**Comment by lievenhey** [comment]
This makes hotspot behave like the user expects. I would be confused if I select multiple cells and only get the last one into my clipboard.

File: src/copyabletreeview.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#pragma once
+
+#include <QTreeView>
+
+class CopyableTreeView : public QTreeView
+{
+    Q_OBJECT public : CopyableTreeView(QWidget* parent = nullptr);
```
Comment (milianw) [inline src/copyabletreeview.h:14]: bad formatting, missing `explicit`

File: src/models/codedelegate.cpp [hunk 2]
```diff
@@ -12,8 +12,7 @@
 #include <QPainter>
 #include <QTextLine>
 
-#include "disassemblymodel.h"
-#include "sourcecodemodel.h"
+Q_DECLARE_METATYPE(QTextLine)
```
Comment (milianw) [inline src/models/codedelegate.cpp:15]: how is this related to the copy operation? separate patch?

**Review by milianw — CHANGES_REQUESTED** [review]

File: src/copyabletreeview.h (hunk 1 above)
Comment (lievenhey) [reply src/copyabletreeview.h:14]: oh wow, apparently this is valid clang formatting

File: src/models/codedelegate.cpp (hunk 2 above)
Comment (milianw) [reply src/models/codedelegate.cpp:15]: can you please answer the question?

**Review by milianw — CHANGES_REQUESTED** [review]

File: src/models/codedelegate.cpp (hunk 2 above)
Comment (lievenhey) [reply src/models/codedelegate.cpp:15]: clangd was complaining the whole time about unused includes so I removed those
